### PR TITLE
Doc cleanup

### PR DESCRIPTION
**Pull request by FrankMittelbach**
# Internal housekeeping

## Status of pull request

- Feedback wanted (maybe something else needs dropping)
- Ready to merge

## Checklist of required changes before merge will be approved
- [n/a] Test file(s) added
- [n/a] Version and date string updated in changed source files
- [n/a] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [ ] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

I would love to autogenerate manifest.md but the current result from `l3build manifest` is not yet really suitable imo.

### CODE REVIEW

**Review by u-fischer — APPROVED** [review]

**Comment by josephwright** [comment]
> I would love to autogenerate manifest.md but the current result from `l3build manifest` is not yet really suitable imo.

Complain at @wspr  ;)

**Review by josephwright — APPROVED** [review]
Do we need the residual files?

**Comment by FrankMittelbach** [comment]
> Do we need the residual files?

not really, but this way they aren't dropped totally while not ending up on ctan. I don't really mind keeping them on github but don't have strong feelings either way. It is just that my experience is that it is nice to keep a trace (and inside the gihub history they are essentially gone for good).

**Comment by FrankMittelbach** [comment]
> > I would love to autogenerate manifest.md but the current result from `l3build manifest` is not yet really suitable imo.
> 
> Complain at @wspr ;)

less a question of complaints :-) more a question of, should we invest in that? We do have to keep a manifest and right now it is likely to be incomplete or out of date (as it is this time around). If we could auto-generate it is more likely to be correct.

**Comment by FrankMittelbach** [comment]
@PhelypeOleinik could you have a look at the .md file? I think all of that is taken care of in which case this file could definitely vanish.

**Review by davidcarlisle — APPROVED** [review]

**Review by zauguin — COMMENTED** [review]
> It is just that my experience is that it is nice to keep a trace (and inside the gihub history they are essentially gone for good).

I'm in favor of dropping these files completely. Especially with titles like "LaTeX installation TeX expert information" they are way too likely to be found by accident when someone is searching for keywords, and then they probably don't even look at some directory name. (That's mostly referring to `texpert.tex` and `tex2.tex`, IMO it makes sense to keep `latexchanges.tex` around for a while)

Beside that: `texpert.txt` and `tex2.txt` are still referenced in the README.md, we should probably drop them there too. `latexchanges.tex` should be removed from `.github/pull_request_template.md`.

**Review by PhelypeOleinik — APPROVED** [review]
In the beginning of `base/README.md`, it says:
> In addition to this bundle, a minimal LaTeX system also needs the files contained in the [list of required packages]

but it doesn't mention `l3kernel`.  I think we should add:

 - L3 programming layer (`l3kernel`)
 - L3 backend interfaces (`l3backend`)

and under `required`:

 - First aid for external packages (`firstaid`)

Other than that, changes seem fine.

**Review by zauguin — APPROVED** [review]

**Comment by muzimuzhi** [comment]
> In the beginning of `base/README.md`, it says:
> 
> > In addition to this bundle, a minimal LaTeX system also needs the files contained in the [list of required packages]
> 
> but it doesn't mention `l3kernel`. I think we should add:
> 
> * L3 programming layer (`l3kernel`)
> * L3 backend interfaces (`l3backend`)
> 
> and under `required`:
> 
> * First aid for external packages (`firstaid`)
> 
> Other than that, changes seem fine.

Maybe also `unicode-data` (for unicode engines) and `l3package` (for `xparse`)?

**Comment by PhelypeOleinik** [comment]
> Maybe also `unicode-data` (for unicode engines) and `l3package` (for `xparse`)?

Maybe `unicode-data`.  `l3packages` is not a dependency (anymore) because the core `xparse` is now part of the LaTeX2e kernel.

**Comment by zauguin** [comment]
`unicode-data` is required (for all engines) by l3kernel, so I think it should be listed somewhere. I agree that it would make sense to add it here, but I think it should also be documented as a requirement in the latex3 repo since it's also needed when expl3 is loaded with other formats.

**Comment by muzimuzhi** [comment]
> > Maybe also `unicode-data` (for unicode engines) and `l3package` (for `xparse`)?
> 
> Maybe `unicode-data`. `l3packages` is not a dependency (anymore) because the core `xparse` is now part of the LaTeX2e kernel.

But `xparse` is stored under `./texmf/tex/latex` repo path, while the mentioned `README.md` is in `./base` (hence readers may think it only has responsible for that subdir?)

**Comment by FrankMittelbach** [comment]
> > 
> 
> But `xparse` is stored under `./texmf/tex/latex` repo path, while the mentioned `README.md` is in `./base` (hence readers may think it only has responsible for that subdir?)

what remains as xparse.sty is *not* required or used by the kernel, it is just another package. so I don't think this is something to list.

**Comment by muzimuzhi** [comment]
> > > 
> > 
> > 
> > But `xparse` is stored under `./texmf/tex/latex` repo path, while the mentioned `README.md` is in `./base` (hence readers may think it only has responsible for that subdir?)
> 
> what remains as xparse.sty is _not_ required or used by the kernel, it is just another package. so I don't think this is something to list.

just found [`./base/ltcmd.dtx`](https://github.com/latex3/latex2e/blob/develop/base/ltcmd.dtx), sorry my bad

**Comment by josephwright** [comment]
> `unicode-data` is required (for all engines) by l3kernel, so I think it should be listed somewhere. I agree that it would make sense to add it here, but I think it should also be documented as a requirement in the latex3 repo since it's also needed when expl3 is loaded with other formats.

Shouldn't that therefore be down as a dependency for `l3kernel` not for LaTeX2e? One normally lists _direct_ dependencies, as otherwise there is the potential to have conflicting information.

**Comment by zauguin** [comment]
> Shouldn't that therefore be down as a dependency for `l3kernel` not for LaTeX2e? One normally lists _direct_ dependencies, as otherwise there is the potential to have conflicting information.

Normally I agree and that's why I suggested that it should be documented for `l3kernel`. But since it's (if I'm not mistaken) the only indirect dependency we currently have, I think that it would be useful to have on complete list here or at least to link to a list of requirements for the other packages. Especially since the current text talks about needing "the files of" `l3kernel` and not needing to install `l3kernel` and all it's dependencies.

I don't have any strong opinions about this though.

**Comment by FrankMittelbach** [comment]
> I don't have any strong opinions about this though.

don't mind mentioning it but I think this is all a  storm in the water glass. nobody these days looks at such a readme to find out what he or she needs to install LaTeX, everybody uses a readymade installation, so all that is more something to drop rather than expand imho.

File: base/README.md [hunk 1]
```diff
@@ -10,10 +10,15 @@ This bundle provides the core LaTeX kernel. In addition to this bundle,
 a minimal LaTeX system also needs the files contained in the
 
 - LaTeX team documentation (`doc`)
+- L3 programming layer (`l3kernel`)
+- L3 backend interfaces (`l3backend`)
+- L3 unicode support (`unicode-data`)
```
Comment (josephwright) [inline base/README.md:15]: Well it's 'core Unicode support' (used by plain XeTeX/LuaTeX), but I'll let you off ;)

**Comment by FrankMittelbach** [comment]
Am 17.11.21 um 17:07 schrieb Joseph Wright:
> +- L3 unicode support (`unicode-data`)
> 
> Well it's 'core Unicode support' (used by plain XeTeX/LuaTeX), but I'll 
> let you off ;)
> 

you're invited to push a change :-)

**Pull request merged**